src/NuGet.Core/NuGet.Packaging/Signing/Utility/VerificationUtility.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using NuGet.Common;
+
+namespace NuGet.Packaging.Signing
+{
+    public class VerificationUtility
+    {
+        internal static bool IsSigningCertificateValid(X509Certificate2 certificate, bool failIfInvalid, List<SignatureLog> issues)
```

Comment:
The "fail" in `failIfInvalid` seems like a bad name; it doesn't change the validation.  Perhaps `treatIssuesAsErrors` or `failuresAreErrors`?